feature/PRO-4188:  new reports domain, fix RT domain & new viewer factory

Daamii opened the pull request:
- Add new Viewer domain to ease integration of map viewer using JS.
  - New API allows to render a map with indoor features and
    control it via JS apis.
  - Allow to show multiple viewers on the same web page.
  - Allow to show realtime users using Situm REST API on top of the
    map viewer
  - Allows to show user trajectories reports using Situm REST API
    on top of the map viewer, among others.
  - Added event subscribing for map is ready, poi selected, building selected
    floor selected
- Add new reports domain that allows to fetch information from the
  Situm REST API /api/v1/reports
- Added biome as new linter
- New tests for reports
- Fixed all tests

albasitum commented on src/domains/realtime.ts:
```diff
@@ -15,7 +15,13 @@ export type ResponseRealtimePosition = {
+  userIds?: UUID[];
+  deviceIds?: string[];
+  indoor?: boolean;
+  max_sec_threshold?: number;
```
`max_sec_threshold` -> `maxSecThreshold`

albasitum commented on src/viewer/index.ts:
esto debería de estar dentro de `domains`? ou no seu módulo a parte?

albasitum commented on src/domains/viewer.ts:
```diff
@@ -0,0 +1,206 @@
+      throw new Error("Viewer iframe not initialized");
+    if (this.realtimeInterval) clearInterval(this.realtimeInterval);
+
+    this.iframe.contentWindow.postMessage(
```
podrías crear un método interno para o envío de mensaxes

albasitum commented on src/domains/viewer.ts:
```diff
@@ -0,0 +1,206 @@
+    }: {
+      buildingIds?: number[];
+    },
+    refreshMs: number = 1000,
```
`refreshRate`en lugar de `refreshMs`?

Daamii replied:
refreshRateMs vale? Gustame que se indiquen as unidades

albasitum commented on src/domains/viewer.ts:
```diff
@@ -0,0 +1,206 @@
+      this.iframe.contentWindow.postMessage(
+        {
+          type: "map.show_trajectory",
+          payload: { speed: 1, status: "PLAY", data: response },
```
cuidado con este "PLAY", eu entendo que pasándolle o data sería suficiente no?

Daamii replied:
decisión interna para automatizar que se empece a reproducir en canto se mande, se se recibe feedback negativo cambiarase

albasitum commented on src/domains/viewer.ts:
```diff
@@ -0,0 +1,206 @@
+      this.iframe.contentWindow.postMessage(
+        {
+          type: "map.show_trajectory",
+          payload: { speed: 1, status: "STOP", data: [] },
```
aquí tamén si lle pasas sólo status y data chegaría entendo eu

Daamii replied:
si, pero bueno tampouco me parece mal reiniciar a velocidad

albasitum commented on examples/viewer-example.html:
```diff
@@ -0,0 +1,97 @@
+  <script src="../dist/umd/situm-sdk.js"></script>
+
+  <script>
+    // Initialize the sdk specifying auth and base domain (later is optional)
```
si non lle pasas o base domain na config en código, vai quedar perdido no comentario creo eu

albasitum commented on examples/viewer-example.html:
```diff
@@ -0,0 +1,97 @@
+      // apiKey: "APIKEY",
+    });
+    viewer.on(ViewerEventType.MAP_IS_READY, () => console.log("viewer1 map is ready"));
+    viewer.on(ViewerEventType.POI_SELECTED, (e) => console.log(e));
```
```
viewer.on(ViewerEventType.POI_SELECTED, (e) => console.log(`Poi with {e.id} was selected`));
```

albasitum commented on examples/viewer-example.html:
```diff
@@ -0,0 +1,97 @@
+
+    let rtFilter = {
+      buildingIds: [BUILDING_ID],
+      // userIds: [],
```
Explicación sobre as cousas que quedan comentadas, p.ej:

"Filter realtime positions by a set of user ids, such that only positions associated to.... "

albasitum commented on examples/viewer-example.html:
```diff
@@ -0,0 +1,97 @@
+      // deviceIds: [] 
+    };
+
+    let trajectoryFilter = {
```
Explicación de que é este filtro, ter 1/2 frases simplemente por aclarar

Daamii commented:
- Add new Viewer domain to ease integration of map viewer using JS.
  - New API allows to render a map with indoor features and
    control it via JS apis.
  - Allow to show multiple viewers on the same web page.
  - Allow to show realtime users using Situm REST API on top of the
    map viewer
  - Allows to show user trajectories reports using Situm REST API
    on top of the map viewer, among others.
  - Added event subscribing for map is ready, poi selected, building selected
    floor selected
- Add new reports domain that allows to fetch information from the
  Situm REST API /api/v1/reports
- Added biome
- New tests for reports
- Fixed all tests
